examples/hybrid_jobs/6_QNSPSA_optimizer_with_embedded_simulator/qnspsa_with_embedded_simulator.ipynb

```diff
@@ -0,0 +1,1360 @@
+      "text/plain": [
+       "<Figure size 640x480 with 1 Axes>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "#intialize a graph for the max cut problem\n",
+    "nodes = n_qubits = 4\n",
+    "edges = 4\n",
+    "seed = 1\n",
+    "\n",
+    "g = nx.gnm_random_graph(nodes, edges, seed=seed)\n",
+    "positions = nx.spring_layout(g, seed=seed)\n",
+    "\n",
+    "#visualize the graph\n",
+    "nx.draw(g, with_labels=True, pos=positions)\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "de84f27d",
+   "metadata": {},
+   "source": [
+    "The QAOA algorithm is an efficient hybrid algorithm solving combinatorial optimization problems such as max cut. In QAOA, finding the solution to the problem is mapped to finding the ground state of a specific Hamiltonian in a variational fashion.\n",
+    "\n",
+    "Pennylane's QAOA module provides support for setting up the max cut problem from a given graph (find detailed information [here](https://pennylane.readthedocs.io/en/stable/code/api/pennylane.qaoa.cost.maxcut.html))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "1ec42556",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Cost Hamiltonian:\n",
+      "  (-2.0) [I0]\n",
+      "+ (0.5) [Z0 Z1]\n",
+      "+ (0.5) [Z0 Z2]\n",
+      "+ (0.5) [Z0 Z3]\n",
+      "+ (0.5) [Z1 Z2]\n",
+      "Cost Hamiltonian:\n",
+      "  (1) [X0]\n",
+      "+ (1) [X1]\n",
+      "+ (1) [X2]\n",
+      "+ (1) [X3]\n"
+     ]
+    }
+   ],
+   "source": [
+    "cost_h, mixer_h = qaoa.maxcut(g)\n",
+    "print(\"Cost Hamiltonian:\")\n",
+    "print(cost_h)\n",
+    "\n",
+    "print(\"Cost Hamiltonian:\")\n",
+    "print(mixer_h)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "73530eaf",
+   "metadata": {},
+   "source": [
+    "Here the expectation of the cost Hamiltonian is bounded to be not greater than 0. Its absolute value (rounded to integer) means how many edges there are between the two subsets of a given cut, while the two subsets are specified by the $Z$ values of each qubit (node). Nodes with the same $Z$ value are of the same subset. Finding the maximum cut is then equivalent to finding the mininum-energy state of the cost Hamiltonian, and this is done with a hybrid algorithm with the QAOA ansatz."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7e076daa",
+   "metadata": {},
+   "source": [
+    "The building block of the QAOA ansatz is a cost layer followed with a mixer layer. This building block repeats for $p$ times, and $p$ is referred as the depth  (or the number of layers) of the ansatz. The cost layer is formed of rotations from the problem Hamiltonian, while the mixer layer is typically chosen to be x-rotation gates on the qubits. This ansatz is visualized in the following cells."
```

Comment:
"This ansatz for **p=2** is visualized..."

Reply:
Let's also not refer to p as the depth. I think that terminology is confusing enough.